Documentation on using Globus for accessing NESE storage in production (#268)
IQSS/dataverse.harvard.edu: doc/globus/upload.md

> # Uploading Data via Globus in the Harvard IQSS Dataverse Archive.

Author: Again, I'm leaning towards "at Harvard Dataverse Archive". Same rational, they are not uploading their stuff to Dataverse, but to this off-site storage facility; Dataverse is only facilitating the upload for them. What I'm trying to convey is _"this is how it works AT our archive here"_

When you start Connect Personal, you will be asked to log in using your Globus account. Verify that your new collection is properly set up by clicking the "Web: Collection Details" in the menu (MacOS version shown):

![Globus Connect Personal Collection View](GlobusPersonalConnect.png)

This should take you to the Globus Web App page, confirming that your personal collection is now part of the Globus federation:

![Globus Web App Collection View](GlobusWebAppCollection.png)

If necessary, add any extra folders containing the data you are depositing to the collection in the _Access_ panel under _Preferences_. Since you are transferring data _from_ your collection, it is safe to leave the "Write Access" box unchecked. The "Shareable" box does not need to be checked either, unless you actually need to make your personal collection accessible to other Globus users.

> ## 3. Borealis Globus App

Reviewer: ```suggestion
## 3. dataverse-globus App
```

> The upload process relies on an extra software component, [dataverse-globus](https://github.com/scholarsportal/dataverse-globus), a web tool that serves as an interface between Dataverse and Globus. It is being developed by Victoria Lubitch at Borealis (formerly Scholars Portal), the Canadian Dataverse Repository project. Clicking the "Upload with Globus" on the _Upload Files_ page in the Dataverse UI will redirect you to the Borealis App in a new tab:

Reviewer: ```suggestion
The upload process relies on an extra software component, [dataverse-globus](https://github.com/scholarsportal/dataverse-globus), a web tool that serves as an interface between Dataverse and Globus. It is being developed by Borealis (formerly Scholars Portal), the Canadian Dataverse Repository project. Clicking the "Upload with Globus" on the _Upload Files_ page in the Dataverse UI will redirect you to the Borealis App in a new tab:
```

![Upload with Globus Button](GlobusUploadButton.png)

(If you are not seeing the "Upload with Globus" button above on the uploads page, that likely means that your Dataverse Collection has not yet been configured to use NESE storage - please contact support to have that straightened out)

> The app will in turn redirect you to a Globus login page (make sure to use an account that has read access to your source collection, in case you have more than one):

Reviewer: ```suggestion
The dataverse-globus app will in turn redirect you to a Globus login page (make sure to use an account that has read access to your source collection, in case you have more than one):
```

![Dataverse Globus App Login](DataverseGlobusAppLogin.png)

The Borealis App will inform you if Globus Connect isn't installed or properly configured on your local system, giving you an option to select a different Globus collection:

![Dataverse Globus App Error](DataverseGlobusAppError.png)

> Otherwise the app will show 2 panels, source on the left, and the destination on the right: 

Reviewer: ```suggestion
Otherwise the app will show two panels, source on the left, and the destination on the right: 
```

![Dataverse Globus App Upload View](DataverseGlobusAppUpload.png)

## 4. Initiating Globus Upload

Navigate to the file you want to upload in the left panel and check the box next to it. It should now appear on the right, in the _destination_ panel: 

![Dataverse Globus App Select File](DataverseGlobusAppFileUploadSelect.png)

then click _Submit Transfer_ at the bottom of the page:

![Dataverse Globus App Submit Transfer](DataverseGlobusAppSubmitUpload.png)

> You should see the "Transfer Initiated" message. You can safely close the Borealis App at this point, since its job is done. The transfer is now happening between the 2 Globus endpoints directly (your instance of Globus Connect Personal needs to be running for the duration of the process; the process can be safely interrupted however - the transfer will restart automatically once the collection is up and running again). The Dataset will remain locked while the transfer is in progress. 

Reviewer: ```suggestion
You should see the "Transfer Initiated" message. You can safely close the dataverse-globus app at this point, since its job is done. The transfer is now happening between the two Globus endpoints directly (your instance of Globus Connect Personal needs to be running for the duration of the process; the process can be safely interrupted however - the transfer will restart automatically once the collection is up and running again). The Dataset will remain locked while the transfer is in progress. 
```

You can use the [Activity](https://app.globus.org/activity) tab of the Globus Web App to monitor the progress. 

> Please keep in mind the experimental and _work in progress_ nature of both the Globus integration support in Dataverse and the Borealis app. Please report to us any problems you may encounter in the process. 

Reviewer: ```suggestion
Please keep in mind the experimental and _work in progress_ nature of both the Globus integration support in Dataverse and the dataverse-globus app. Please report to us any problems you may encounter in the process. 
```
